### DISCUSSION TITLE
Would it be useful for Linux distributions iso?

### DISCUSSION THREAD

**Discussion by bigbruno** [mhx/dwarfs#135] [General]
Has a Linux iso ever been made using Dwarfs instead of Squashfs? Do you think it would be advantageous?

**Comment by mhx** [comment]
> Has a Linux iso ever been made using Dwarfs instead of Squashfs?

I'm not aware of any. If you come across one, definitely let me know.

> Do you think it would be advantageous?

I think it depends. For "regular" distro, you probably won't get much gain over SquashFS. OTOH, if we're talking about a live ISO with a large amount of highly redundant data (e.g., coming back to my original use case, a live ISO with 1000+ versions of Perl installed), then yeah, absolutely.

**Reply by krasnh** [reply]
>For "regular" distro, you probably won't get much gain over SquashFS.

Is this still true? Even for "dwarfs-0.12.1"?

By the way, some distributions transfer LiveCD to EROFS. For example, fedora 42, which will be released on April 15th.

**Reply by mhx** [reply]
> > For "regular" distro, you probably won't get much gain over SquashFS.
> 
> Is this still true? Even for "dwarfs-0.12.1"?

Maybe it wasn't even true back when I made the comment! :)

There's [another thread](https://github.com/mhx/dwarfs/discussions/250#discussioncomment-12533481) where someone is trying to look into creating an live image that uses dwarfs.

There are also [some comparisons](https://github.com/mhx/dwarfs/discussions/250#discussioncomment-12380138) of the image sizes.

**Comment by mhx** [comment]
I've moved this to discussions for now as it doesn't really seem to be an issue.

**Reply by godane** [reply]
I added support to my custom linux-live scripts here: https://github.com/godane/linux-live/commit/3595532e67a9f3eed5e1ed7a6dc964b770db13f7

It works very well and i have done over 100 modules of dwarfs. Just know that i think each module takes up about 4mb of ram i think. I was mostly using this with media like pdfs and videos. The pdfs of some stuff did save a few gigabytes more of space vs squashfs with lzo. 
PS I added these as options for dwarfs modules in later commit: -o readonly -o cachesize=32M

**Reply by krasnh** [reply]
@godane, almost two years have passed since your message, can you share new feedback on your use of dwarfs in Slax?
And if you share a working iso for checking from a flash drive, that would be absolutely wonderful. )
I looked at the official iso (https://ftp.sh.cvut.cz/slax/), I thought your edits got there, but no.

I also use a distribution with a similar modular design in the style of Puppy Linux, with squashfs.

**Reply by mhx** [reply]
> Just know that i think each module takes up about 4mb of ram i think.

As mentioned already on #154, I'm planning to work on support for mounting multiple DwarFS images using a single process and using a shared cache. This should significantly reduce the memory overhead. This is tracked by #219.
